Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs

```diff
@@ -150,23 +172,23 @@ public static string CreateUnexpectedResponseCodeMessage(HttpStatusCode status,
                                                       string asset = null,
                                                       string account = null)
         {
-            if (TestExceptionForMessage(e, "No such host is known", asset))
+
+            if (TestExceptionForMessage(e, s_noSuchHostIsKnown, ref asset))
             {
                 return ReturnUnknownHost(ref message, asset);
             }
 
-            if (TestExceptionForMessage(e, "The network path was not found", asset))
+            if (TestExceptionForMessage(e, s_networkPathNotFound, ref asset))
             {
                 return ReturnUnknownHost(ref message, asset);
             }
 
-            if (TestExceptionForMessage(e, "The remote name could not be resolved", asset))
+            if (TestExceptionForMessage(e, s_remoteNameCouldNotBeResolved, ref asset))
```

Comment:
>s_remoteNameCouldNotBeResolved [](start = 43, length = 30)

This regex will retrieve the remote name from the exception in cases where the provided asset is null.